crates/typst-library/src/layout/mod.rs

```diff
@@ -4,6 +4,7 @@ mod abs;
 mod align;
 mod angle;
 mod axes;
+pub mod cellgrid;
```

Comment:
FWIW, if you agree with my proposal of using `ResolvedGrid` and moving to `typst-library/src/layout/grid/{mod.rs, resolve.rs}`, I'd be ok with doing it in this PR as it's a (comparatively) pretty minor change, so you can focus on your proposed refactors on the other PR.

Reply:
I think that it's better to keep the `CellGrid` name and structure for now, to keep this PR minimally invasive. For later, I think that it might be nice to have a module `typst-library/src/layout/resolved/`, where `CellGrid` and other resolved types could live, if there are enough to justify this. But that is just an idea for now.

Reply:
Sure. What do you think about at least changing to `grid/{mod.rs, resolve.rs}`, otherwise keeping `CellGrid` as is? I really dislike having something like `typst-library/layout/cellgrid.rs`, at least imo.

Reply:
OK, I did this, @PgBiel. However, because we have `pub use self::grid::*;` in `typst-library/layout/mod.rs`, this means that we now expose `layout::resolved`. We can change this by doing `pub use self::grid::{Grid, GridItem, ...};` instead, but this is a bit verbose (because `grid` contains quite some types), and no other module in `layout` exports its items explicitly. What do you think?

Reply:
Yeah, this is basically the goal of keeping it separate, as we don't need to use the `resolve` module in many places, only in typst-layout/grid basically.

Also, not sure if that was intentional but you named the module `resolved`, but I think `resolve` is more appropriate as the main functions it exports convert grids, tables and cells to their resolved counterparts. Do you think you could rename it? Then I think there won't be anything else to change.